pkg/skaffold/deploy/status_check.go

```diff
@@ -112,7 +113,7 @@ func getDeployments(client kubernetes.Interface, ns string, l *DefaultLabeller,
 	deployments := make([]Resource, 0, len(deps.Items))
 	for _, d := range deps.Items {
 		var deadline time.Duration
-		if d.Spec.ProgressDeadlineSeconds == nil {
+		if d.Spec.ProgressDeadlineSeconds == nil || *d.Spec.ProgressDeadlineSeconds == kubernetesMaxDeadline {
```

Comment:
doesn't this mean that if someone sets up their deadline to 600, we'll just override it? 
can we have a test for this?

Reply:
unfortunately yes.  When creating the object from `yaml`, the default will be set to 600 if not specified. 
We will end up overriding in case user specifies 600 as limit.

Reply:
done. We already have tests where deadline is not set.

Reply:
I think this might be surprising for users so we should document it! can you add a note to to the docs?

Reply:
Just got hit by this.... ;)

Reply:
😬  uh oh...this seems like a bug to me. can't we check if the user explicitly specified this deadline and then break out of this logic? that's not a great UX.